src/main/java/gregtech/common/covers/CoverPump.java

```diff
@@ -134,10 +135,8 @@ public void update() {
     }
 
     protected int doTransferFluids(int transferLimit) {
-        PooledMutableBlockPos blockPos = PooledMutableBlockPos.retain();
-        blockPos.setPos(coverHolder.getPos()).move(attachedSide);
-        TileEntity tileEntity = coverHolder.getWorld().getTileEntity(blockPos);
-        blockPos.release();
+        BlockPos pos = coverHolder.getPos().offset(attachedSide);
```

Comment:
Worth it to invest a `MutablePos` object, this gets called in potentially every tick update.